lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -1267,11 +1321,102 @@ def test_enroll_with_email_not_registered_autoenroll(self, protocol):
+                'course_mode': CourseMode.VERIFIED,
+            },
+            **environ
+        )
+        self.assertEqual(response.status_code, 200)
+
+        # expected response data for students update enrollment api
+        expected = {
+            'action': 'enroll',
+            'auto_enroll': True,
+            'results': [
+                {
+                    'identifier': self.notregistered_email,
+                    'before': {
+                        'enrollment': False,
+                        'auto_enroll': False,
+                        'user': False,
+                        'allowed': False,
+                    },
+                    'after': {
+                        'enrollment': False,
+                        'auto_enroll': True,
+                        'user': False,
+                        'allowed': True,
+                    }
+                }
+            ]
+        }
+
+        # check that the student is not yet enrolled in the 'verified' mode
+        # instead of enrollment there in new record in the model
+        # 'CourseEnrollmentAllowed' with desired course mode
+        self.assertEqual(
+            CourseEnrollment.objects.filter(user=self.notenrolled_student, course_id=self.course.id).count(), 0
+        )
+        self.assertTrue(
+            CourseEnrollmentAllowed.objects.filter(email=self.notregistered_email, course_id=self.course.id).exists()
+        )
+        self.assertEqual(
+            CourseEnrollmentAllowed.objects.get(email=self.notregistered_email, course_id=self.course.id).mode,
+            CourseMode.VERIFIED
+        )
+
+        # check that the student is not yet enrolled in the 'verified' mode
+        # verify the enrollment api response data
+        manual_enrollments = ManualEnrollmentAudit.objects.all()
+        self.assertEqual(manual_enrollments.count(), 1)
+        self.assertEqual(manual_enrollments[0].state_transition, UNENROLLED_TO_ALLOWEDTOENROLL)
+        res_json = json.loads(response.content)
+        self.assertEqual(res_json, expected)
+
+        # Check the outbox and verify the content of generated email
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertEqual(
+            mail.outbox[0].subject,
+            u'You have been invited to register for {}'.format(self.course.display_name)
+        )
+        self.assertEqual(
+            mail.outbox[0].body,
+            "Dear student,\n\nYou have been invited to join {display_name} at edx.org by a member of the "
+            "course staff.\n\n"
+            "To finish your registration, please visit {protocol}://{site}/register and fill out the "
+            "registration form making sure to use {user_email} in the E-mail field.\n"
+            "Once you have registered and activated your account, "
+            "you will see {display_name} listed on your dashboard.\n\n----\n"
+            "This email was automatically sent from edx.org to {user_email}".format(
```

Comment:
@srpearce We will need a separate ticket to update the email templates since changing the email content was not part of the story.
FYI: @douglashall